ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js

```diff
@@ -222,10 +222,17 @@ module.exports = Marionette.LayoutView.extend({
     this.filterAttribute.currentView.model.set('value', [attribute])
   },
   handleAttributeUpdate: function() {
+    const previousAttributeType =
+      metacardDefinitions.metacardTypes[this.model.get('type')].type
     this.model.set(
       'type',
       this.filterAttribute.currentView.model.get('value')[0]
     )
+    const currentAttributeType =
```

Comment:
```suggestion
    handleAttributeUpdate: function() {
    const previousAttributeType = metacardDefinitions.metacardTypes[this.model.get('type')].type
    const nextAttributeType = this.filterAttribute.currentView.model.get('value')[0]
    this.model.set({
        type: nextAttributeType,
        ...(previousAttributeType !== nextAttributeType) && { value: ['']}
    });
    }
```

Reply:
I'm not sure that will compare the types I had in mind. Here, `previousAttributeType` will be something like `STRING` or `LOCATION`, but `nextAttributeType` will be something like `anyText` or `anyGeo`

Reply:
Ah, I see, the declaration of the nextAttributeType should be:
```
 const nextAttributeType = metacardDefinitions.metacardTypes[this.filterAttribute.currentView.model.get('value')[0]].type
```